crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -67,9 +84,248 @@ pub fn is_participation_output(output: &Output) -> bool {
+                                // a candidate and continue with the iterator
+                                if destinations.contains_tokens_timelocked_target() {
+                                    self.ordered_timelock_candidates.insert(
+                                        self.ordered_timelock_candidates
+                                            .binary_search_by(|(_, output)| {
+                                                timelock_ordering(output, &inner)
+                                            })
+                                            .unwrap_or_else(|e| e),
+                                        (header.clone(), inner.clone()),
+                                    );
+                                    continue;
+                                }
+                            } else {
+                                // if it is just a basic output, try to perform the SwapSplit
+                                // operation for several destinations once all tokens targets are
+                                // meet.
+                                let (possibly_modified_original_output_maybe, split_outputs_maybe) =
+                                    swap_split_operation(
+                                        destinations.iter_by_tokens_target_mut_filtered(),
+                                        basic_output,
+                                    );
+                                // if some SwapSplit were performed, their result are basic inputs
+                                // stored in split_outputs_maybe; so, we save them in
+                                // split_basic_outputs to return them later
+                                if split_outputs_maybe.len() > 0 {
+                                    self.num_swapped_basic += 1;
+                                }
+                                self.split_basic_outputs.extend(
+                                    split_outputs_maybe
+                                        .into_iter()
+                                        .map(|output| (header.clone(), output)),
+                                );
+                                // if there was a remainder, the original output is returned for the
+                                // iterator, possibly with a modified amount; else, continue the
+                                // loop
+                                if let Some(possibly_modified_original_output) =
+                                    possibly_modified_original_output_maybe
+                                {
+                                    *inner = possibly_modified_original_output;
+                                } else {
+                                    continue;
+                                }
+                            };
+                        }
+                    }
+                }
+                return Some(output);
+            }
```

Comment:
This is what I am referring to above:
```suggestion
            }
            return Some(output);
```